pallets/transfer-allowlist/src/lib.rs

```diff
@@ -76,32 +84,23 @@ pub mod pallet {
 	pub trait Config: frame_system::Config {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
-		/// Id type of Currency transfer restrictions will be checked for
-		type CurrencyId: AssetId
-			+ Parameter
-			+ Debug
-			+ Default
-			+ Member
-			+ Copy
-			+ MaybeSerializeDeserialize
-			+ Ord
-			+ TypeInfo
-			+ MaxEncodedLen;
+		type CurrencyId: AssetId + Parameter + Member + Copy + Default;
+
+		// We need this to block restrictions on the native chain currency as
+		// currently it is impossible to place transfer restrictions
+		// on native currencies as we simply have no way of
+		// restricting pallet-balances...
+		type NativeCurrency: Get<Self::CurrencyId>;
```

Comment:
I need to look deeper into this pallet so this is more a question for myself but doesn't our `pallet-restricted-transfers` take care of this for us? Why would we need to hope pallet-balances to offer that for us here?

Reply:
but users can always use `pallet-balances` to transfer CFG and we can not not exposte the `Call` for balances as this breaks all integrations with external parties. Unfortunately...